msteams-platform/samples/virtual-tables-api.md

```diff
@@ -0,0 +1,229 @@
+
+> [!NOTE]
+> Currently Collaboration controls are available only in [public developer preview](~/resources/dev-preview/developer-preview-intro.md).
+
+## Virtual table sorting
+
+With the virtual tables, you can use the OData $orderby query parameter to set criteria for how the result set should be sorted. Use the asc or desc suffix to specify ascending or descending order respectively. The default is ascending if the suffix isn't applied.  
+
+**Supported Tables**: Each virtual table supports the same sorting functionality as it’s respective Graph resource. The virtual tables, which support sorting are:  
+
+* Graph Drive Item
+* Graph Event
+
+> [!NOTE]
+> Sorting is not supported on all the attributes of the respective Graph resources. If a user tries to sort on a virtual table with an unsupported attribute, this result set will have its default order. This is the same behaviour as the Dataverse Web API on columns that do not support sorting.
+
+Examples:
+
+* GET [Organization URI]/api/data/v9.2/m365_graphdriveitems?$filter=m365_collaborationrootid eq ‘00000000-0000-0000-0000-000000000000’&$orderby=m365_name desc
+* GET [Organization URI]/api/data/v9.2/m365_graphevents?$filter=m365_groupid eq ‘00000000-0000-0000-0000-000000000000’$orderby=m365_subject asc
+
+## Virtual table filtering
+
+With the virtual tables, you can use the OData $filter query parameter to set criteria for which rows will be returned. The virtual tables are queried using the same OData operators that are supported by the Dataverse Web API.
+
+* **Comparison operators**
+
+  |Operator|Description|Example|
+  |----|----|----|
+  |eq|Equal|$filter=m365_name eq ‘Contoso’|
+  |ne|Not Equal|$filter=m365_name ne ‘Contoso’|
+  |gt|Greater Than|$filter=m365_price gt 50.0|
+  |ge|Greater Than or Equal|$filter=m365_price ge 50.0|
+  |lt|Less Than|$filter=m365_price lt 50.0|
+  |le|Less Than or Equal|$filter=m365_price le 50.0|
+
+* **Logical operators**
+
+  |Operator|Description|Example|
+  |----|----|----|
+  |and|Logical and |$filter=m365_name eq ‘Contoso’ and m365_price eq 50.0|
+  |or|Logical or |$filter=m365_name ne ‘Contoso’ or m365_price eq 50.0|
+  |not|Logical negotiation |$filter=not contains(m365_name,’Contoso’)|
+
+* **Grouping operators**
+
+  |Operator|Description|Example|
+  |----|----|----|
+  |( )|Precedence grouping |$filter=(m365_name eq ‘Contoso’ and m365_price eq 50.0) or contains(m365_subject,’Team Sync’)|
+
+* **Query Functions**
+
+  |Function |Example |
+  |----|----|
+  |contains|$filter=contains(m365_name,’Contoso’)|
+  |endswith|$filter=endswith(m365_name,’Contoso’)|
+  |startswith|$filter=startswith(m365_name,’Contoso’)|
+
+**Supported Tables**: Each virtual table supports the same filtering functionality as it’s respective Graph resource. The virtual tables, which support filtering are:
+
+* Graph Booking Appointment
```

Comment:
Under both the Sorting section above and the Filtering section here we list the table names with their display names e.g. `Graph Booking Appointment` further down though in the Pagination section the tables are presented with their non-display names e.g. `m365_graphbookingappointment`. Should this be made consistent throughout the doc?